cirq-core/cirq/ops/control_values.py

```diff
@@ -0,0 +1,175 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import abc
+from typing import Union, Tuple, List, TYPE_CHECKING, Any, Dict, Generator, Optional, cast, Iterator
+from dataclasses import dataclass
+
+import itertools
+
+if TYPE_CHECKING:
+    import cirq
+
+
+class AbstractControlValues(abc.ABC):
+    """AbstractControlValues is an abstract immutable data class.
+
+    AbstractControlValues defines an API for control values and implements
+    functions common to all implementations (e.g.  comparison).
+    """
```

Comment:
The docstring should be more detailed. For example:

```suggestion
    """Abstract immutable base class that defines the API for control values.

    `cirq.ControlledGate` and `cirq.ControlledOperation` are useful to augment existing gates
    and operations to have one or more control qubits. For every control qubit, the set of
    integer values for which the control should be enabled is represented by
    `cirq.AbstractControlValues`.

    Implementations of `cirq.AbstractControlValues` can use different internal representations
    to store control values, but they should all satisfy the public API defined here.
    """
```

Reply:
PTAL